src/forms/Slushbucket.tsx

```diff
@@ -0,0 +1,237 @@
+  }
+
+  handleInput(e: any): void {
+    this.setState({ searchString: e.target.value });
+  }
+
+  render() {
+    const {
+      choices,
+      value,
+      isFetching,
+      placeholder,
+      valueChanged
+    } = this.props;
+
+    const selected = value;
+    const updateModelValue = valueChanged;
+
+    return (
+      <div className={`${styles.SelectChoice} form-input`}>
+        <div
+          style={{
+            width: "50%",
+            display: "flex",
+            justifyContent: "space-between"
+          }}
+        >
+          <input
+            tabIndex={-1}
+            type="text"
+            autoComplete="false"
+            className={
+              formStyles.FormControl +
+              " " +
+              (this.props.readonly ? inputStyles.ReadOnly : "")
+            }
+            placeholder={placeholder}
+            onChange={e => this.handleInput(e)}
+            onKeyUp={event => {
+              this.handleKeyUp(event);
+            }}
+            value={this.state.searchString}
+            disabled={this.props.readonly}
+            readOnly={this.props.readonly}
+          />
+          <div
+            className={`${styles.Spinner} ${isFetching
+              ? displayStyles.Block
+              : displayStyles.None}`}
+          >
+            <MDSpinner size={18} />
+          </div>
+
+          <div
+            style={{
+              transform: "translateX(-33px)",
+              display: "flex",
+              alignItems: "center",
+              visibility: this.state.searchString === "" ? "hidden" : "visible"
+            }}
+          >
+            <ClearButton
+              onClick={() => {
+                this.resetSearchString();
+              }}
+            />
+          </div>
+        </div>
+
+        <div
+          style={{
+            display: "flex"
+          }}
+        >
+          <div
+            className={styles.Results}
+            style={{
+              marginRight: "20px"
+            }}
+          >
+            <div className={`${styles.SelectedRow}`}>
+              <b>Available</b>
+            </div>
+            <Scrollbars autoHeight autoHeightMin={400} autoHeightMax={400}>
+              {choices
+                .filter(choiceItem => {
+                  if (this.props.readonly) {
+                    return false;
+                  }
+                  if (this.state.searchString === "") {
+                    // if nothing is typed show all results
+                    return true;
+                  } else {
+                    // if user typed search string only show those that contain string
+                    // TODO sort by search string ?
+                    return choiceItem
+                      .display
+                      .toLowerCase()
+                      .includes(this.state.searchString.toLowerCase());
+                  }
+                })
+                .map((choiceItem, i) => {
```

Comment:
If you put .map(({value, display}, i) => { here then it saves a lot of choiceItem. later. Not sure if it works and is better, but still.

Reply:
Good point. Did it and works